cmake/module/AddBoostIfNeeded.cmake

```diff
@@ -54,5 +55,54 @@ function(add_boost_if_needed)
     endif()
   endif()
 
-  mark_as_advanced(Boost_INCLUDE_DIR)
+  if(WITH_EXTERNAL_SIGNER)
+    include(CheckCXXSourceCompiles)
+    set(CMAKE_REQUIRED_INCLUDES ${Boost_INCLUDE_DIR})
+    # Boost 1.78 requires the following workaround.
+    # See: https://github.com/boostorg/process/issues/235
+    set(CMAKE_REQUIRED_FLAGS "-Wno-error=narrowing")
+    # Boost 1.73 requires the following workaround on systems with
+    # libc<2.34 only, as newer libc has integrated the functionality
+    # of the libpthread library.
+    set(CMAKE_REQUIRED_LIBRARIES Threads::Threads)
+    check_cxx_source_compiles("
+      #define BOOST_PROCESS_USE_STD_FS
+      #include <boost/process.hpp>
+      #include <string>
+
+      int main()
+      {
+        namespace bp = boost::process;
+        bp::opstream stdin_stream;
+        bp::ipstream stdout_stream;
+        bp::child c(\"dummy\", bp::std_out > stdout_stream, bp::std_err > stdout_stream, bp::std_in < stdin_stream);
+        stdin_stream << std::string{\"test\"} << std::endl;
+        if (c.running()) c.terminate();
+        c.wait();
+        c.exit_code();
+      }
+      " HAVE_BOOST_PROCESS
+    )
+    if(HAVE_BOOST_PROCESS)
+      if(WIN32)
```

Comment:
NIT: if it's WIN32 couldn't you just return early with external signer OFF and not bother checking if boost supports boost process? Would cleanup one nested if.

Reply:
You are right. However, I've chosen this clumsy approach to mimic the Autotools' one for the sake of easier reviewing,

Anyway, this code will gone after https://github.com/bitcoin/bitcoin/pull/28981.